### DISCUSSION TITLE
[Feature] Add AV1 support for video previews

### DISCUSSION THREAD

**Discussion by bt90** [immich-app/immich#2211] [Feature Request]
### The feature

We currently support h264 and hevc as video codecs for previews. It would be awesome to have also support for av1.

The reasoning is quite simple. It offers the same benefits as hevc but is also supported on Firefox.

https://caniuse.com/av1

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by truongsinh** [comment]
I took a quick look, and it seems that:
- Container image must be upgraded from alpine 3.14 to 3.17, so that ffmpeg can be upgraded to the version supporting AV1
- FFMPEG, when running as `ffmpeg -i <input_file>  -vcodec av1 <output_file>`, will use libaom-av1, which is extremely slow, we want to run  `ffmpeg -i <input_file>   -c:v libsvtav1 <output_file>` instead
- The preset of AV1/libsvtav1 is numerical, unlike string of h264

**Reply by bo0tzz** [reply]
The alpine update is being worked on in #2242. It seems like after that, the rest should be fairly straightforward?

**Reply by truongsinh** [reply]
WDYT about the latter 2 points (using `libsvtav1` instead of default `libaom-av1`, and preset is numerical, not something like `ultrafast`)? In general, the problem is that, for each codec (or even same codec with different lib), the params can be different, and it is hard to consolidate them in a nice list of dropdown menu.

One more thing, I read somewhere that AV1 compression really shines in (high resolution AND low bitrate) combination. I.e we might even need to guide user that if they want to use AV1, the recommended setting is to not limit the resolution at all.

**Reply by zackpollard** [reply]
I think these are potentially sensible decisions, I would need to do some more research into AV1 encoding recommendations before we added it into immich however. I don't see a reason why we wouldn't add this as an option, but I think there are some other transcoding improvements I would like to make before we start adding more options into the mix. I'll make a note to revisit this once I have finished with the other improvements 🙂

**Reply by AngelaDMerkel** [reply]
There is very little hardware decode or encode support for AV1 and serving AV1 would be quite taxing for most systems. Despite that, AV1 support would be great insofar as animated AVIF is likely to catch on vis a vis gif and gifv (h264 in and mp4 container).

**Comment by n00mkrad** [comment]
Any news on this?

There's plenty of AV1 encoding and decoding hardware around nowadays (all recent GPUs) and even software-based SVT-AV1 (encoder) and libdav1d (decoder) are very fast.

**Reply by patrontheo** [reply]
Yes, that would be awesome !
@zackpollard do you know if this in the plans ? 
Thanks

**Reply by zackpollard** [reply]
I don't think we are against adding AV1 support to the transcoder, someone did just open a PR for it. The complexity is getting all the current options we have working with new codec options.

**Reply by patrontheo** [reply]
Cool, thanks for the answer :)
The reason I asked is because I use an iPhone (doesn't support VP9), and Firefox (doesn't support HEVC), so I have to fallback to h.264 (or use another browser).
EDIT: I realize now that I'm not sure the iPhone will be able to play AV1, but let's see, and it's still a nice feature to have :)
EDIT2: I'll probably use HEVC and a webapp on my mac to view the videos ([mac web apps](https://support.apple.com/en-us/104996)), instead of firefox.

**Reply by sutusa** [reply]
@patrontheo The latest chip announcements from Apple stated hardware-level AV1 decode.  The [iPhone A17 chips](https://www.apple.com/newsroom/2023/09/apple-unveils-iphone-15-pro-and-iphone-15-pro-max/) & [Mac M3 chips](https://www.apple.com/nz/newsroom/2023/10/apple-unveils-m3-m3-pro-and-m3-max-the-most-advanced-chips-for-a-personal-computer/).  I believe the latest iPhones and Macs selling right now have those chips.

**Reply by patrontheo** [reply]
@sutusa  Yes, that's also what I saw ! I don't have the latest iPhone though, but good they added it ! :)

**Comment by bo0tzz** [comment]
#8491
